Sources/NostrSDK/Events/TextNoteEvent.swift

```diff
@@ -131,66 +136,99 @@ public extension EventCreating {
     ///
     /// See [NIP-01](https://github.com/nostr-protocol/nips/blob/master/01.md)
     /// See [NIP-10 - On "e" and "p" tags in Text Events (kind 1)](https://github.com/nostr-protocol/nips/blob/master/10.md)
+    @available(*, deprecated, message: "Deprecated in favor of TextNote.Builder.")
     func textNote(withContent content: String, replyingTo repliedEvent: TextNoteEvent? = nil, mentionedEventTags: [EventTag]? = nil, subject: String? = nil, customEmojis: [CustomEmoji]? = nil, signedBy keypair: Keypair) throws -> TextNoteEvent {
```

Comment:
As a maintainer, there's no need to worry about adding to a trainwreck of a function if we need to add new functionality. 

As a developer using the SDK, I can now customize the specific event kinds in my client with the `NostrEvent.Builder`, especially if the SDK hasn't caught up to implementing the latest NIP updates. It gives me flexibility to create my own event, or combine my own schema modifications to what the SDK provides.